### PR TITLE
SC-88445: add-code-coverage-to-all-dx-projects

### PR DESCRIPTION
**Pull request by neo-anand-patel**
## Details
Added code coverage

## Additional context
Configured GitHub workflow to add code coverage

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -28,4 +30,15 @@ jobs:
       - name: Build
         run: yarn build
       - name: Test
-        run: yarn test
+        run: yarn test -- --coverage
+      - name: Upload coverage
+        uses: codecov/codecov-action@v1
+        with:
+          token: ${{secrets.CODECOV_TOKEN}}
```
Comment (neo-anand-patel) [inline .github/workflows/ci.yaml:37]: getting a warning in build, maybe this token isn't set up? Not sure.

File: README.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 [![Build status](https://github.com/neofinancial/mongo-memory-server-action/workflows/ci/badge.svg)](https://github.com/neofinancial/mongo-memory-server-action/actions)
 [![Test status](https://github.com/neofinancial/mongo-memory-server-action/workflows/pull-request/badge.svg)](https://github.com/neofinancial/mongo-memory-server-action/actions)
 [![Publish status](https://github.com/neofinancial/mongo-memory-server-action/workflows/publish/badge.svg)](https://github.com/neofinancial/mongo-memory-server-action/actions)
+[![coverage](https://coverage.neotools.ca/api/badge/master/4EUt9DTZZpn3u1KfBtNpjn)](https://coverage.neotools.ca/coverage/neofinancial/mongo-memory-server-action)
```
Comment (neo-anand-patel) [inline README.md:6]: Not sure where to get the link for the Codecov badge.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (ghost) [reply .github/workflows/ci.yaml:37]: You will see that the "test" script in package.json already has `--coverage` built into it, and I suspect that's the problem you encountered.

The workflows for this GitHub Action repository are designed for it to be able to publish its own tags when PRs are pushed to the mainline branch, avoiding the need to do this manually. To ensure the coverage is reported similarly to other code-bases, the Coverage step should be added to the `pull-request` workflow. I've got a separate PR open for this - this one could probably be closed and the branch removed.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md:6]: That link provides the badge :+1: (see [this reference page](https://github.com/dwyl/repo-badges))

**Pull request closed without merging**